### ISSUE TITLE
Cannot parse CREATE INDEX statement on a field called "field"

### ISSUE THREAD

**Issue by JanecekPetr** [jOOQ/jOOQ#7568]
### Expected behavior and actual behavior:

The following DDL kills the parser:

```sql
CREATE TABLE foo (
    bbb BIGSERIAL,
    field TEXT NOT NULL,
    PRIMARY KEY (bbb, field)
);
CREATE INDEX foo_field_idx ON foo (field);
```

with the following message:

```
Token '(' expected: [6:41] ...eld)
);
CREATE INDEX foo_field_idx ON foo (field[*]);
```

where the problem apparently is the `field` column. This works:

```sql
CREATE TABLE foo (
    bbb BIGSERIAL,
    fielda TEXT NOT NULL,
    PRIMARY KEY (bbb, fielda)
);
CREATE INDEX foo_fielda_idx ON foo (fielda);
```

and so does this:

```sql
CREATE TABLE foo (
    bbb BIGSERIAL,
    "field" TEXT NOT NULL,
    PRIMARY KEY (bbb, "field")
);
CREATE INDEX foo_field_idx ON foo ("field");
```

so I'm thinking we may have missed an obvious SQL reserved word or something. That said, I cannot find "field" in any SQL reserved word list which is why we used it in the first place. Is there any rule we have missed, or is this a jOOQ parser bug?

### Steps to reproduce the problem (include example code if possible):

Run the DDL code above through the jOOQ parser.

### Versions:

- jOOQ: 3.11
- Java: N/A
- Database (include vendor): The DDL has been written for any is being accepted by Postgres
- OS: N/A
- JDBC Driver (include name if inofficial driver): N/A

**Comment by lukaseder** [comment]
Thanks a lot for your report.

This is a parser bug. `field()` is a MySQL function, and the parser complains about the function being incomplete. I would imagine that this happens with other functions as well.

In general, the jOOQ parser shouldn't complain about reserved words because there's always one dialect that would parse the keyword as an identifier. Having to resort to `"quoting"` the "keywords" is a workaround, but not an acceptable solution.

I'll further investigate this to see what other functions may be affected.

Thanks again for your report.

**Comment by lukaseder** [comment]
This really only affects the name "field", which is being parsed as a keyword, not a function name

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.12. Backport to 3.11.2 scheduled: #7607.
